skimage/feature/blob.py

```diff
@@ -92,6 +92,13 @@ def _prune_blobs(blobs_array, overlap):
     # iterating again might eliminate more blobs, but one iteration suffices
     # for most cases
     for blob1, blob2 in itt.combinations(blobs_array, 2):
+
+        # Ignore if we have removed either blob during an earlier iteration.
+        if blob1[3] == -1:
```

Comment:
But I've just realised that it should actually be 

if (blob1[2] == -1) or (blob2[2] == -1)):

Because the 'radius' is at index 2. Apologies.